### DISCUSSION TITLE
geo load balancing and locality based load balancing

### DISCUSSION THREAD

**Discussion by raffaelespazzoli** [k8gb-io/k8gb#366] [Ideas]
There are two features that a global load balancer should probably have to cover enterprise use cases and they are:
1. geo load balancing: given the IP of the caller, determined the closest endpoint based on geographical instance. The assumption  is that the ip can be used to locate the geographical point of origin of the request and that geographical distance approximates network latency.
2. locality based load balancing is when the load balancer decision is made on some latency metrics known to be true. Typical use case: when a company has two datacenters callas from datacenter A should be directed to endpoints in datacenerte A preferably.
I'd like to start designing how these two features can be added to k8gb. Some considerations:
1. these load balancing policy do not exist in coredns, new plugin will have to be created.
2. these new plugin will need to make the decision based on a database of metrics and locality metadata of the endpoints:
   a. we can assume that the database of metrics will be passed as part of the configuration.
   b. I am not sure what we can do for the locality metadata of the endpoints.

**Comment by donovanmuller** [comment]
@raffaelespazzoli please feel free to comment on anything missing from #244 which captures most of the requirements above.
